rust/hyperlane-base/src/types/s3_storage.rs

```diff
@@ -176,4 +181,18 @@ impl CheckpointSyncer for S3Storage {
         .await?;
         Ok(())
     }
+
+    async fn write_announcement(&self, signed_announcement: &SignedAnnouncement) -> Result<()> {
+        let serialized_announcement = serde_json::to_string_pretty(signed_announcement)?;
+        self.write_to_bucket(S3Storage::announcement_key(), &serialized_announcement)
+            .await?;
+        Ok(())
+    }
+    fn announcement_metadata(&self) -> String {
```

Comment:
could maybe use `format!` instead for the String formatting. Here & local storage